### PR TITLE
quote all the shims and remove unncessary /usr/bin/env

### PR DESCRIPTION
**Pull request by joshuarli**
f/u to https://github.com/getsentry/devenv/pull/142

properly quotes everything in shims and removes unnecessary /usr/bin/env!

### CODE REVIEW

File: devenv/lib/node.py [hunk 1]
```diff
@@ -126,7 +127,8 @@ def install_yarn(version: str, reporoot: str) -> None:
     fs.write_script(
         f"{binroot}/yarn",
         f"""#!/bin/sh
-exec /usr/bin/env PATH={binroot}/node-env/bin:"$PATH" {binroot}/node-env/bin/yarn "$@"
+export PATH="{binroot}/node-env/bin:${{PATH}}"
```
[the remark this answers is not in the document]
Comment (joshuarli) [reply devenv/lib/node.py:130]: `exec: FOO=1: not found`

```
#!/bin/sh
exec FOO=1 bash -c 'env | grep FOO'
```

File: devenv/lib/gcloud.py [hunk 2]
```diff
@@ -27,13 +27,17 @@ def _install(url: str, sha256: str, into: str) -> None:
     fs.write_script(
         f"{into}/gcloud",
         f"""#!/bin/sh
-exec /usr/bin/env CLOUDSDK_PYTHON={root}/python/bin/python3 PATH={into}/google-cloud-sdk/bin:$PATH gcloud "$@"
+export CLOUDSDK_PYTHON="{root}/python/bin/python3" \
+       PATH="{into}/google-cloud-sdk/bin:${{PATH}}"
```
Comment (asottile-sentry) [inline devenv/lib/gcloud.py:31]: technically we should be using `shlex.quote` on all of these values (protect against someone `mkdir '$(cat /etc/passwd)'`)

Comment (joshuarli) [reply devenv/lib/gcloud.py:31]: should i make write_script better by passing an optional format dict that'll then shell escape all those tokens?

Comment (asottile-sentry) [reply devenv/lib/gcloud.py:31]: if it makes it simpler sure that sounds decent

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**